File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(n);
        int totalSum = 0;
        for (int num : nums) totalSum += num;
        int prefixSum = 0;
        for (int i = 0; i < n; i++) {
            int left = nums[i] * i - prefixSum;
            int right = (totalSum - prefixSum - nums[i]) - nums[i] * (n - i - 1);
            result[i] = left + right;
            prefixSum += nums[i];
        }
        return result;
    }
};

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        
        int totalSum = 0;
        for (int num : nums) {
            totalSum += num;
        }
        
        int leftSum = 0;
        
        for (int i = 0; i < n; i++) {
            int rightSum = totalSum - leftSum - nums[i];
            
            int leftPart = nums[i] * i - leftSum;
            int rightPart = rightSum - nums[i] * (n - i - 1);
            
            result[i] = leftPart + rightPart;
            
            leftSum += nums[i];
        }
        
        return result;
    }
}